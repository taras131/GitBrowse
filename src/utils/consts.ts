export const REPOS = {
    TITLE: "Репозитории:",
    SEARCH_PLACEHOLDER: "Поиск Репозиториев",
    ITEMS_PER_PAGE: 20,
    SORT_BUTTON_DESK_TEXT: "Сначала новые",
    SORT_BUTTON_ASK_TEXT: "Сначала старые",
    NOT_DESCRIPTION_TEXT: "Нет описания",
    LINK_TEXT: "Перейти",
    EMPTY_TITLE: "У пользователя нет репозиториев.",
    EMPTY_TEXT: "Попробуйте изменить фильтр или создать новый репозиторий.",
    HISTORY_SUBTITLE: "История:",
} as const;

export const INFORM_MESSAGE = {
    START: "Введите не менее трёх символов для начала поиска",
    USER_FOUND: "Пользователь найден",
    USER_NOT_FOUND: "К сожалению , пользователь не найден",
    ERROR: "При загрузке возникла ошибка",
    INVALID_USERNAME: "Недопустимое имя пользователя",
} as const;

export const START_SEARCH_HISTORY = ["gaearon", "markerikson", "taras131"];
export const SORT = {ASK: "asc", DESC: "desc"};
export type TSort = (typeof SORT)[keyof typeof SORT];