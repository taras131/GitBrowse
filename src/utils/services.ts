export const formatDate = (dateString: string) => {
  const date = new Date(dateString);
  return date.toLocaleDateString("ru-RU", {
    year: "numeric",
    month: "2-digit",
    day: "numeric",
  });
};