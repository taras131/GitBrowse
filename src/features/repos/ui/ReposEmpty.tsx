import React, { memo } from "react";
import Typography from "@mui/material/Typography";
import { Warning as WarningIcon } from "@mui/icons-material";
import Card from "@mui/material/Card";
import { REPOS } from "../../../utils/consts";

const CARD_STYLES = {
  card: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    height: "100%",
    p: 4,
    gap: 2,
  },
  warning_icon: {
    fontSize: 40,
    color: "warning.main",
    mb: 1,
  },
  subtitle_1: {
    fontWeight: "bold",
  },
};

const ReposEmpty = memo(() => {
  return (
    <Card sx={CARD_STYLES.card}>
      <WarningIcon data-testid="WarningIcon" sx={CARD_STYLES.warning_icon} />
      <Typography variant="subtitle1" sx={CARD_STYLES.subtitle_1} textAlign="center">
        {REPOS.EMPTY_TITLE}
      </Typography>
      <Typography variant="body2" color="text.secondary">
        {REPOS.EMPTY_TEXT}
      </Typography>
    </Card>
  );
});

export default ReposEmpty;
