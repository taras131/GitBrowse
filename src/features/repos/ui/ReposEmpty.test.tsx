import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReposEmpty from "./ReposEmpty";
import { REPOS } from "../../../utils/consts";

describe("ReposEmpty Component", () => {
  beforeEach(() => {
    render(<ReposEmpty />);
  });

  test("renders warning icon", () => {
    const warningIcon = screen.getByTestId("WarningIcon");
    expect(warningIcon).toBeInTheDocument();
  });

  test("renders main message about no repositories", () => {
    const mainMessage = screen.getByText(REPOS.EMPTY_TITLE);
    expect(mainMessage).toBeInTheDocument();
    expect(mainMessage.tagName).toBe("H6");
    expect(mainMessage).toHaveClass("MuiTypography-subtitle1");
  });

  test("renders secondary message with suggestions", () => {
    const secondaryMessage = screen.getByText(
      REPOS.EMPTY_TEXT
    );
    expect(secondaryMessage).toBeInTheDocument();
    expect(secondaryMessage).toHaveClass("MuiTypography-body2");
  });

  test("card has correct styles", () => {
    const card = screen.getByText(REPOS.EMPTY_TEXT).closest(".MuiCard-root");
    expect(card).toBeInTheDocument();
    expect(card).toHaveClass("MuiCard-root");
    expect(card).toHaveStyle({
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    });
  });
  test("matches snapshot", () => {
    const { container } = render(<ReposEmpty />);
    expect(container).toMatchSnapshot();
  });
});