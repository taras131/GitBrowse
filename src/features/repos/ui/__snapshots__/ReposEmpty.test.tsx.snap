// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposEmpty Component matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1nuf7y8-MuiPaper-root-MuiCard-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-g1fzfi-MuiSvgIcon-root"
      data-testid="WarningIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M1 21h22L12 2zm12-3h-2v-2h2zm0-4h-2v-4h2z"
      />
    </svg>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-1ms7rs1-MuiTypography-root"
    >
      У пользователя нет репозиториев.
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Попробуйте изменить фильтр или создать новый репозиторий.
    </p>
  </div>
</div>
`;
